setup/vsim.sh

```diff
@@ -1,2 +1,6 @@
-ROOT=$(cd "$(dirname "${BASH_SOURCE[0]:-$0}")/.." && pwd)
+if [ -n "$ZSH_VERSION" ]; then
+    ROOT=$(cd "$(dirname "${(%):-%N}")/.." && pwd)
+else
+    ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+fi
 export VSIM_PATH="$ROOT"/sim
```

Comment:
You don't need to change this file. The `${BASH_SOURCE[0]:-$0}` with the `:` makes it so that it works under `zsh` already.